src/runtime/runtime.ml

```diff
@@ -663,6 +674,10 @@ let rec equal_value v1 v2 =
        (* XXX should we compare references by value instead? *)
        Store.Ref.key_eq v1 v2
 
+    | Dyn v1, Dyn v2 ->
+       (* XXX should we compare dynamics by value instead? *)
```

Comment:
No. OCaml compares refs by value and it's annoying and unexpected. I think of dynamics as refs.